File: Apps/iOS7/Bounds/Bounds/BoundsViewController.m
//
//  BoundsViewController.m
//  Bounds
//
//  Created by Clemens Wagner on 22.06.14.
//  Copyright (c) 2014 Clemens Wagner. All rights reserved.
//

#import "BoundsViewController.h"

@interface BoundsViewController()

@property (nonatomic, weak) IBOutlet UIView *boundsView;

@property (nonatomic, weak) IBOutlet UISlider *xSlider;
@property (nonatomic, weak) IBOutlet UISlider *ySlider;

@property (nonatomic, weak) IBOutlet UILabel *xLabel;
@property (nonatomic, weak) IBOutlet UILabel *yLabel;

@end

@implementation BoundsViewController

- (void)viewWillAppear:(BOOL)inAnimated {
    [super viewWillAppear:inAnimated];
    [self updateX];
    [self updateY];
}

- (IBAction)updateX {
    UIView *theView = self.boundsView;
    CGRect theFrame = theView.frame;
    CGRect theBounds = theView.bounds;
    CGFloat theWidth = CGRectGetWidth(theFrame);
    
    theBounds.origin.x = self.xSlider.value * theWidth;
    theView.bounds = theBounds;
    theView.frame = theFrame;
    self.xLabel.text = [NSString stringWithFormat:@"%.f", CGRectGetMinX(theBounds)];
    NSLog(@"frame: %@", NSStringFromCGRect(theView.frame));
}

- (IBAction)updateY {
    UIView *theView = self.boundsView;
    CGRect theFrame = theView.frame;
    CGRect theBounds = theView.bounds;
    CGFloat theHeight = CGRectGetHeight(theFrame);
    
    theBounds.origin.y = self.ySlider.value * theHeight;
    theView.bounds = theBounds;
    theView.frame = theFrame;
    self.yLabel.text = [NSString stringWithFormat:@"%.f", CGRectGetMinY(theBounds)];
    NSLog(@"frame: %@", NSStringFromCGRect(theView.frame));
}

@end

File: Apps/iOS8/Swift/Bounds/Bounds/BoundsViewController.swift
//
//  BoundsViewController.swift
//  Bounds
//
//  Created by Clemens Wagner on 22.06.14.
//  Copyright (c) 2014 Clemens Wagner. All rights reserved.
//

import UIKit

class BoundsViewController: UIViewController {
    @IBOutlet var boundsView: UIView
    @IBOutlet var xSlider: UISlider
    @IBOutlet var ySlider: UISlider
    
    @IBOutlet var xLabel: UILabel
    @IBOutlet var yLabel: UILabel

    override func viewWillAppear(inAnimated: Bool) {
        super.viewWillAppear(inAnimated)
        updateX()
        updateY()
    }
    
    func updateX() {
        let theView = self.boundsView
        let theFrame = theView.frame
        let theWidth = theFrame.width
        var theBounds = theView.bounds
        
        theBounds = CGRect(x: self.xSlider.value * theWidth, y: theBounds.minY, width: theBounds.width, height: theBounds.height)
        theView.bounds = theBounds
        theView.frame = theFrame
        self.xLabel.text = NSString(format:"%.f", theBounds.minX)
    }
    
    func updateY() {
        let theView = self.boundsView
        let theFrame = theView.frame
        let theHeight = CGRectGetHeight(theFrame)
        var theBounds = theView.bounds
        
        theBounds = CGRect(x: theBounds.minX, y: self.ySlider.value * theHeight, width: theBounds.width, height: theBounds.height)
        theView.bounds = theBounds
        theView.frame = theFrame
        self.yLabel.text = NSString(format:"%.f", theBounds.minY)
    }
}
